unstructured_ingest/v2/processes/connectors/github.py

```diff
@@ -0,0 +1,264 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from datetime import datetime
+from time import time
+from typing import TYPE_CHECKING, Any, Generator, Optional
+from urllib.parse import urlparse
+
+from pydantic import Field, Secret, root_validator
+
+from unstructured_ingest.error import SourceConnectionError, SourceConnectionNetworkError
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+    download_responses,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+CONNECTOR_TYPE = "github"
+if TYPE_CHECKING:
+    from github.Repository import Repository
+
+
+class GitHubAccessConfig(AccessConfig):
+    git_access_token: Optional[str] = Field(
+        default=None,
+    )
+
+
+class GitHubConnectionConfig(ConnectionConfig):
+    url: str
+    access_config: Secret[GitHubAccessConfig]
+    branch: Optional[str] = Field(default=None, overload_name="git_branch")
+    repo_path: str = field(init=False, repr=False, default=None)
+
+    @root_validator(pre=True)
+    def set_repo_path(cls, values):
+        # Parse the URL
+        url = values.get("url")
+        if url:
+            parsed_gh_url = urlparse(url)
+            path_fragments = [fragment for fragment in parsed_gh_url.path.split("/") if fragment]
+
+            # Validate the URL and construct the repo_path
+            if (
+                (parsed_gh_url.scheme and parsed_gh_url.scheme != "https")
+                or (parsed_gh_url.netloc and parsed_gh_url.netloc != "github.com")
+                or len(path_fragments) != 2
+            ):
+                raise ValueError(
+                    'Please provide a valid URL, e.g. "https://github.com/owner/repo" or '
+                    '"owner/repo".'
+                )
+
+            # Set the repo_path based on URL fragments
+            values["repo_path"] = "/".join(path_fragments)
+        return values
+
+    @SourceConnectionError.wrap
+    @requires_dependencies(["github"], extras="github")
+    def get_repo(self) -> "Repository":
+        from github import Github
+
+        github = Github(self.access_config.get_secret_value().git_access_token)
+        return github.get_repo(self.repo_path)
+
+
+class GitHubIndexerConfig(IndexerConfig):
+    pass
+
+
+@dataclass
+class GitHubIndexer(Indexer):
+    connection_config: GitHubConnectionConfig
+    index_config: GitHubIndexerConfig
+
+    @staticmethod
+    def is_file_type_supported(path: str) -> bool:
+        # Workaround to ensure that auto.partition isn't fed with .yaml, .py, etc. files
+        # TODO: What to do with no filenames? e.g. LICENSE, Makefile, etc.
+        supported = path.endswith(
```

Comment:
We should probably remove this seeing as the partition logic is always expanding to support more file types. Let the partitioner handle the case of the file type not being handled.